### DISCUSSION TITLE
copilot token chat_enabled: false

### DISCUSSION THREAD

**Discussion by justinmchase** [community/community#68463] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I have installed the Copilot extension, I can see it working in the panel. I have access through an Enterprise and an org. 

But the chat seems to not work. I can't see the panels and when I look in the Output panel at Copilot chat I see the following logs:

```
2023-09-27T19:20:27.438Z [INFO] [auth] Logged in as jchase24_uhg
2023-09-27T19:20:27.951Z [INFO] [chat] copilot token chat_enabled: false
2023-09-27T19:20:27.952Z [INFO] [auth] Got Copilot token for jchase24_uhg
```

I'm assuming its not working because of `chat_enabled: false`. How can I toggle it to `chat_enabled: true`?

**Comment by wweevv-johndpope** [comment]
![Screenshot from 2023-09-28 11-50-50](https://github.com/community/community/assets/109928647/cb130d12-76da-4c56-a2be-fdbd3025ebb0)

not sure if this is going to get anywhere. but technically speaking....

a few lines later 
there's 1:0 - I just hack to 1:1

start from here
![Screenshot from 2023-09-28 11-57-20](https://github.com/community/community/assets/109928647/3ea0e1a0-7a52-491d-8341-e617e291292a)

![Screenshot from 2023-09-28 11-54-59](https://github.com/community/community/assets/109928647/6d635d5e-17bb-40e5-97b0-5117e15828ba)

(I'm yet to see it work)
![Screenshot from 2023-09-28 11-58-00](https://github.com/community/community/assets/109928647/d9d42d95-d786-407b-9b3f-d6acbebd88b6)


![Screenshot from 2023-09-28 11-58-23](https://github.com/community/community/assets/109928647/32da837a-f2ac-4ac3-9e2a-ec1a87d2dba4)

**Reply by technorainbows** [reply]
This worked for me!

**Comment by Primus66** [comment]
It seems like you're encountering an issue with the Copilot extension for Visual Studio Code, where the chat functionality is disabled (chat_enabled: false). To enable the chat feature, you might need to perform a few troubleshooting steps:

- Check Copilot Version: Ensure that you're using the latest version of the Copilot extension. Outdated versions may have bugs or lack certain features.

- GitHub Copilot Subscription: Make sure that your GitHub Copilot subscription includes chat support. Depending on your subscription level (e.g., free trial, individual subscription, or part of an organization plan), chat functionality may or may not be available.

- Extension Updates: Keep an eye on updates for the Copilot extension. Sometimes, enabling chat functionality may require an update to the extension itself.

- Always ensure that you're using the latest version of the extension and that you're properly authenticated with the required permissions to access chat functionality.

**Comment by wweevv-johndpope** [comment]
https://docs.github.com/en/copilot/github-copilot-chat/using-github-copilot-chat

![IMG_8273](https://github.com/community/community/assets/109928647/f811392c-90cf-4229-be69-e666cd8c58b1)


I don’t see the setting in my GitHub. - I bought the licenses. Where is the enable chat beta ?

**Comment by wweevv-johndpope** [comment, marked as answer]
Update - 

https://github.com/organizations/YOURORG/settings/copilot/policies

![IMG_8275](https://github.com/community/community/assets/109928647/a762c5b2-cbae-470d-a9ef-6c16e1bdb6f9)
![IMG_8274](https://github.com/community/community/assets/109928647/40b43f8e-8475-4cd4-874e-02e6d1917523)

**Reply by justinmchase** [reply]
So the answer is the setting needs to be enabled on the org, thank you!

**Comment by davecheney** [comment]
@justinmchase you need to visit https://github.com/github-copilot/free_signup to claim you free copilot access. Chat is included.

**Reply by justinmchase** [reply]
It's enterprise, that seems to not work for enterprise accounts.

Also the question is about copilot chat specifically. I'm already signed up for copilot, but the chat is still not enabled.

**Reply by johandanforth** [reply]
Got the same problem, also via enterprise account:

copilot token chat_enabled: false

**Reply by DaniSancas** [reply]
I have the same issue:

I have the enterprise license. And it doesn't work in my case. 

```
2023-10-17T08:36:03.889Z [INFO] [auth] Logged in as DaniSancas
2023-10-17T08:36:04.412Z [INFO] [chat] copilot token chat_enabled: false
2023-10-17T08:36:04.413Z [INFO] [auth] Got Copilot token for DaniSancas
```

Is there any way to report it and get actual help from Github Copilot devs?

**Reply by wweevv-johndpope** [reply]
Enable it in settings as per screenshot above - it’s the only way

**Reply by mikkorantalainen** [reply]
I'll comment here to hopefully help Googling this page in the future.

I had a problem that I could use Copilot in VS Code but Copilot Chat didn't show up anywhere even though I used action "Github Copilot Chat" in the command palette. There were no visible error messages or warnings anywhere.

When I finally found View – Output – "Github Copilot Chat", I could see message

2023-11-07T16:48:35.990Z [INFO] [chat] copilot token chat_enabled: false

which finally allowed me to find this discussion and figure out that the problem was actually policy set in GitHub organization Copilot settings.

It would have been great to display a notice/info/warning about this when I tried to use the Copilot Chat commands in VS Code so this is a silent failure. It appeared as if the UI would have been opened somewhere (because I didn't have any feedback or errors) but the panel with the chat would be too small or scrolled off the screen. In reality, the action that's available from GitHub icon in bottom left simply fails silently if the policy is set incorrectly!

To fix the policy (if you are organization admin in GitHub) you have to do following:

- Open the organization front page in GitHub (URL should look like http://github.com/NameOfYourOrganization)
- Click "Settings" on top navigation
- Click Code, planning, and automation – Copilot – Policies and features
- Look for "GitHub Copilot Chat for IDEs" in Features. For some reason, this defaulted to Disabled for our organization and was the root cause for the issue.
